src/main/java/com/TheAccountant/service/impl/LoanNotificationService.java

```diff
@@ -0,0 +1,59 @@
+package com.TheAccountant.service.impl;
+
+import com.TheAccountant.dao.LoanDao;
+import com.TheAccountant.dao.NotificationDao;
+import com.TheAccountant.model.loan.Loan;
+import com.TheAccountant.model.notification.Notification;
+import com.TheAccountant.model.notification.NotificationCategory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import java.sql.Timestamp;
+import java.util.List;
+
+/**
+ * Created by tudor.grigoriu on 3/19/2017.
+ */
+public class LoanNotificationService {
+
+    private static final String PREFIX = "Loan active! ";
+    private static final String TO_RECEIVE_FROM = "To receive from ";
+    private static final String TO_GIVE_TO = "To give to ";
+    @Autowired
+    private LoanDao loanDao;
+
+    @Autowired
+    private NotificationDao notificationDao;
+
+    @Scheduled(cron = "0 0 0 * * *") //everyday at midnight
+    public void addLoanNotifications(){
+        Timestamp currentTimestamp = new Timestamp(System.currentTimeMillis());
+        List<Loan> loanList = loanDao.findAllActiveBeforeDate(currentTimestamp);
```

Comment:
I suggest that the Loan entity should have a field 'until date' that will be used for notification if that date is smaller than the current date, and the loan is still active, in that case the priority for the notification should be HIGH..